Add Jobs for Product Creation, Builds, and Publishing

FyreByrd opened the pull request:
Handles the creation of a Product in BuildEngine, as well as the Build and Release processes in BuildEngine.
Properly handles the creation of ProductArtifacts in the database.
Also added UI for creating a Product.

![image](https://github.com/user-attachments/assets/22ac0dca-d2ff-46d9-b66d-c082d732ea7d)

![image](https://github.com/user-attachments/assets/d9f7ee7f-fe4f-46a2-b691-af39d9bb01e2)

![image](https://github.com/user-attachments/assets/4511658e-9f27-41a2-865d-ea76dfe08770)

![image](https://github.com/user-attachments/assets/97b5f320-669b-42ed-ab8e-59b36e5fec91)

FyreByrd commented:
A few things I am not certain on.
1. Do the modal boxes for product creation need a visible cancel button?
2. How should StoreLanguages be handled?

I am also experiencing some weird bugs with BuildEngine right now.

For these reasons, this is currently a draft PR.

FyreByrd commented:
One other thing that should be resolved first: What does the expired status mean and how should it be handled?

FyreByrd commented:
I resolved the BuildEngine issue, so this should be ready for review/requests

chrisvire commented:
> 1. Do the modal boxes for product creation need a visible cancel button?

S1 has a close X in the top right.

FyreByrd commented:
> > 1. Do the modal boxes for product creation need a visible cancel button?
> 
> S1 has a close X in the top right.

Added

chrisvire commented:
## Select Product to Add
User Experience:
1. Light mode is using Dark mode coloring for the description for each product (and selecting store in the next dialog)

![image](https://github.com/user-attachments/assets/76b4d4c3-6a00-4fbc-b342-af56c3650366)

![image](https://github.com/user-attachments/assets/7a2bbd6d-8f46-4310-8d89-6607be72f173)

## App Builder Configuration
User Experience:
1. We have instructions for copying the **App Project URL**.  With DWKit, we didn't have an easy way to allow the user to copy the URL. Should we have a copy icon at the end text field to make it easy to copy the contents?
2. In Light mode, the action buttons ("New App" and "Exisiting App" buttons)  are a little hard to see.

![image](https://github.com/user-attachments/assets/e4d49d6e-855f-4942-b638-ec5272e2cb03)

Here is S1

![image](https://github.com/user-attachments/assets/f43fadb9-e784-4093-9a9f-236435ace9b3)

FyreByrd commented:
@chrisvire I fixed the visibility issues and added a copy icon

![image](https://github.com/user-attachments/assets/07f8244a-5b51-45c9-adad-8a12a052ec7d)

![image](https://github.com/user-attachments/assets/49191c26-5d14-44fc-9fd7-14cafc62a477)

FyreByrd commented on scripts/DB/default_workflow.sql:
I made these changes because of how I was handling `ProductType`.
The four product types: `Android_GooglePlay`, `Android_S3`, `AssetPackage`, `Web`, were the different product types I was able to derive based on the forms in DWKit. (Granted, the schemes in DWKit are slightly different; there is a distinct one for PWA, and not for Asset Package).

Since this was the set of product types that I went with, I eventually realized that I needed slightly more information from somewhere in order to properly build products. Since I didn't have a separate product type for PWA, I added that as to `WorkflowDefinitions.Properties` for the relevant workflow definitions. Since I also no longer needed that same information for the asset packages, I likewise removed it.

chrisvire commented:
How are you handling building PWA vs. Modern PWA ?
With S1, there is a "Modern PWA to Cloud" product definition which uses the pwa_cloud workflow and pwa_could_rebuild rebuild workflow and set the properties to be:

```json
{
  "build:targets" : "modern-pwa"
}
```

FyreByrd commented:
> How are you handling building PWA vs. Modern PWA ? With S1, there is a "Modern PWA to Cloud" product definition which uses the pwa_cloud workflow and pwa_could_rebuild rebuild workflow and set the properties to be:
> 
> ```json
> {
>   "build:targets" : "modern-pwa"
> }
> ```

I tested with a product definition using those properties. It should work now.

chrisvire reviewed (approved):
Thanks for all the good work!
